manifests/dropin_file.pp

```diff
@@ -50,9 +50,11 @@
 
   if $ensure != 'absent' {
     ensure_resource('file', "${path}/${unit}.d", {
-      ensure => directory,
-      owner  => 'root',
-      group  => 'root',
+      ensure  => directory,
+      owner   => 'root',
+      group   => 'root',
+      recurse => $::systemd::purge_dropin_dirs,
```

Comment:
Is recursing needed? I thought purge would be enough.

Reply:
@ekohl I think recurse is needed, at least that's my read of docs for both purge and recurse.  I use that pattern in my profiles where I'm essentially doing what this define does, it works as expected though recurse does seem unnecessary at first.

https://docs.puppet.com/puppet/latest/type.html#file-attribute-purge

Reply:
this is maybe a good candidate for an acceptance test?